xonsh/commands_cache.py

```diff
@@ -18,6 +18,11 @@
 from xonsh.platform import ON_POSIX, ON_WINDOWS, pathbasename
 from xonsh.tools import executables_in
 
+if ON_WINDOWS:
```

Comment:
This behavior could be unified too if xonsh wanted to take a position on case-sensitivity (it could be case-sensitive on all platforms or case-insensitive on all platforms).

Reply:
I don't think it's obvious that the change should apply to macOS. While it's true that on Windows and Mac, the default file systems are case-insensitive, all operating systems offer file systems that can be either case sensitive or not.

Probably what we really want is to have the _lookup_ be case insensitive, but then additionally verify that the file exists by the name given. The real challenge arises when a file exists under two or more names varying only by case. In that scenario, the caching wouldn't work unless it mapped to all possible values (i.e. a CaseInsensitive MultiDict).

I was thinking that xonsh could be case sensitive or case insensitive on all operating systems, but if it's case insensitive on a case-sensitive filesystem, it will not be able to reach filenames that vary only by case, and if it's case sensitive on case-insensitive filesystems, users might be surprised when a file is only reachable by the exact case.

The latter option (being case-sensitive on all platforms) seems the most safe and tolerable, especially if all we're talking about is command matching.

Regardless, I don't want to try to solve that issue here. In this PR, I'm focused on the more narrow issue of name mangling that was happening in the command cache on Windows. If someone is interested in willing to explore the issue in more depth, I welcome them to file an issue.